Sources/Support/POSIX/ELFSupport.cpp

```diff
@@ -161,5 +163,70 @@ bool ELFSupport::ReadSectionHeader(int fd, const ELFHeader &ehdr, SectionHeader
 
   return true;
 }
+
+template <typename ELFHeader, typename SectionHeader>
+bool ELFSupport::ReadStringTable(int fd, const ELFHeader &ehdr,
+                                 SectionHeader &shdr,
+                                 std::vector<char> &table) {
+  // e_shstrndx is the index of the string table containing section names
+  if (!ReadSectionHeader(fd, ehdr, shdr, ehdr.e_shstrndx))
+    return false;
+
+  if (shdr.sh_type != SHT_STRTAB)
+    return false;
+
+  table.resize(shdr.sh_size);
+  if (::pread(fd, &table[0], shdr.sh_size, shdr.sh_offset) != shdr.sh_size)
+    return false;
+
+  return true;
+}
+
+template <typename ELFHeader, typename SectionHeader>
+bool ELFSupport::ReadDebugLinkCRC(int fd, const ELFHeader &ehdr,
+                                  SectionHeader &shdr, ByteVector &crc) {
+
+  std::vector<char> table;
+  if (!ReadStringTable(fd, ehdr, shdr, table))
+    return false;
+
+  for (size_t i = 0; i < ehdr.e_shnum; i++) {
+    if (!ReadSectionHeader(fd, ehdr, shdr, i))
+      return false;
+
+    if (shdr.sh_type != SHT_PROGBITS)
+      continue;
+
+    const std::string sectionName(&table[shdr.sh_name]);
+    if (sectionName != ".gnu_debuglink")
```

Comment:
I wonder if we should create a global constant for the section name.

Reply:
I will. That was lazy.